libvast/vast/system/actors.hpp

```diff
@@ -222,8 +224,8 @@ using index_actor = typed_actor_fwd<
   // Conform to the protocol of the STATUS CLIENT actor.
   ::extend_with<status_client_actor>::unwrap;
 
-using archive_request
-  = std::tuple<caf::typed_response_promise<atom::done>, receiver<table_slice>>;
+using archive_request = std::tuple<caf::typed_response_promise<atom::done>,
+                                   receiver<table_slice>, expression>;
```

Comment:
Move the expression and the promise into the `unhandled_ids` map, and remove the early popping.

Reply:
The archive state management has been simplified and this comment does not apply any more.